srtcore/srt_c_api.cpp

```diff
@@ -98,13 +98,13 @@ int srt_rendezvous(SRTSOCKET u, const struct sockaddr* local_name, int local_nam
     // Just as a safety precaution, check the structs.
     if ( (local_name->sa_family != AF_INET && local_name->sa_family != AF_INET6)
             || local_name->sa_family != remote_name->sa_family)
-        return SRT_EINVPARAM;
+        return CUDT::APIError(MJ_NOTSUP, MN_INVAL, 0);
```

Comment:
As you are doing this already, you might also add a hook on the above call to `CUDT::setsockopt`. This function may also fail if you pass an invalid or connected socket to it. Simply forward the returned -1 - the error code is set already.

Reply:
Invalid socket will failed in srt_bind anyway.

Reply:
Exactly.

Reply:
Well, ok. :)